primitives/reshape.fil

```diff
@@ -67,21 +74,31 @@ comp Serialize[#W, #N, #B, ?#C=1, ?#H=1]<G: #C*(#N-1)+#H>(
 // In general, you probably don't want to use the #C and #H parameters at
 // all. They are most useful when the deserializer is consuming inputs from
 // a component that is not fully pipelined.
-comp Deserialize[#W, #N, #B, ?#C=1, ?#H=1]<G: #C*#N+#H-2>(
+comp Deserialize[#W, #N, #B, ?#C=1, ?#H=1]<G: #End-1>(
```

Comment:
The usage of `let` bindings made it obvious that we were holding onto the signals longer than we needed to!